nix-tools/make-install-plan/MakeInstallPlan.hs

```diff
@@ -0,0 +1,128 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+import qualified Cabal2Nix
+import qualified Data.ByteString.Lazy as BSL
+import Data.Foldable (for_)
+import Distribution.Client.DistDirLayout (DistDirLayout (distDirectory, distProjectCacheFile, distProjectFile))
+import Distribution.Client.GlobalFlags
+import Distribution.Client.HashValue (HashValue, showHashValue)
+import qualified Distribution.Client.InstallPlan as InstallPlan
+import Distribution.Client.NixStyleOptions (NixStyleFlags (..), defaultNixStyleFlags, nixStyleOptions)
+import Distribution.Client.ProjectConfig
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.ProjectPlanOutput (writePlanExternalRepresentation)
+import Distribution.Client.ProjectPlanning (ElaboratedConfiguredPackage (..), rebuildInstallPlan)
+import Distribution.Client.Setup
+import Distribution.Client.Types.PackageLocation (PackageLocation (..))
+import Distribution.Client.Types.Repo (LocalRepo (..), RemoteRepo (..), Repo (..))
+import Distribution.Client.Types.SourceRepo (SourceRepositoryPackage (..))
+import Distribution.Package (pkgName)
+import Distribution.Pretty (prettyShow)
+import Distribution.Simple.Command
+import Distribution.Simple.Flag
+import qualified Distribution.Simple.Utils as Cabal
+import Distribution.Types.SourceRepo (KnownRepoType (Git), RepoType (..))
+import Distribution.Verbosity (Verbosity)
+import qualified Distribution.Verbosity as Verbosity
+import Nix.Pretty (prettyNix)
+import Prettyprinter (Doc)
+import Prettyprinter.Render.Text (hPutDoc)
+import System.Environment (getArgs)
+import System.FilePath
+import System.IO (IOMode (WriteMode), hClose, openFile)
+
+main :: IO ()
+main = do
+  args <- getArgs
+  case commandParseArgs cmdUI True args of
+    CommandHelp help -> putStrLn (help "something")
+    CommandList opts -> putStrLn $ "commandList" ++ show opts
+    CommandErrors errs -> putStrLn $ "commandErrors: " ++ show errs
+    CommandReadyToGo (mkflags, _commandParse) ->
+      let globalFlags = defaultGlobalFlags
+          flags@NixStyleFlags {configFlags} = mkflags (commandDefaultFlags cmdUI)
+          verbosity = fromFlagOrDefault Verbosity.normal (configVerbosity configFlags)
+          cliConfig = commandLineFlagsToProjectConfig globalFlags flags mempty
+       in installPlanAction verbosity cliConfig
+
+cmdUI :: CommandUI (NixStyleFlags ())
+cmdUI =
+  CommandUI
+    { commandName = "",
+      commandSynopsis = "Makes an install-plan",
+      commandUsage = ("Usage: " ++),
+      commandDescription = Nothing,
+      commandNotes = Nothing,
+      commandDefaultFlags = defaultNixStyleFlags (),
+      commandOptions = nixStyleOptions (const [])
+    }
+
+installPlanAction :: Verbosity -> ProjectConfig -> IO ()
```

Comment:
I'm guessing you nicked a lot of this logic from cabal, worth a pointer to where in cabal this sort of thing happens for future reference?